source/Engine.cpp

```diff
@@ -1554,7 +1555,13 @@ void Engine::CalculateStep()
 	bool wasHyperspacing = (flagship && flagship->IsEnteringHyperspace());
 	// First, move the player's flagship.
 	if(flagship)
+	{
+		emptySoundsTimer.resize(flagship->Weapons().size());
+		for(auto &it : emptySoundsTimer)
```

Comment:
I tend to only use `auto` when it's replacing a long/complex object type (e.g. an iterator). But in this case, you're replacing `int`, which is actually shorter than `auto`. Might as well just make it clear that these are integers.
```suggestion
		for(int &it : emptySoundsTimer)
```